media_gateway_server/src/server/service/crypto/argon2.rs

```diff
@@ -0,0 +1,55 @@
+use anyhow::bail;
+use argon2::{Argon2, PasswordHash, PasswordVerifier};
+
+use crate::server::service::crypto::PasswordService;
+
+pub struct Argon2PasswordService {}
+
+impl PasswordService for Argon2PasswordService {
+    fn verify(&self, password: &str, password_hash: &str) -> anyhow::Result<bool> {
+        match PasswordHash::new(&password_hash) {
```

Comment:
This will give
> error[E0277]: the trait bound `argon2::password_hash::Error: StdError` is not satisfied
 --> media_gateway_server/src/server/service/crypto/argon2.rs:9:60
  |
9 |         let parsed_hash = PasswordHash::new(&password_hash)?;
  |                                                            ^ the trait `StdError` is not implemented for `argon2::password_hash::Error`, which is required by `Result<bool, anyhow::Error>: FromResidual<Result<Infallible, argon2::password_hash::Error>>`
  |

Works like
```
        let parsed_hash = PasswordHash::new(&password_hash)
            .map_err(|e| anyhow::anyhow!("Error while parsing password hash: {:?}", e))?;
        Ok(Argon2::default()
            .verify_password(password.as_bytes(), &parsed_hash)
            .is_ok())
```